File: renamebytime.m
function renamebytime(file_path, file_extension)
    if nargin == 1
        file_extension = '.JPG';
    end

    nowfolder = [file_path, '\*', file_extension];
    img = dir(nowfolder);
    
    for i = 1:length(img)
        % Get the origin name and create the new name.
        origin_name = img(i).name;
        date = datetime(img(i).date, 'InputFormat', 'dd-MMMM-yyyy HH:mm:ss', 'Locale', 'zh_CN');
        filename = char(datetime(date, 'Format', 'MMdd_HHmmss'));
        new_name = [filename, '.JPG'];
       
        % Check if filename repeats
        a = 1;
        while isfile(new_name)
            new_name = [filename, '_', a, '.JPG'];
            a = a + 1;
        end

        % '!'的意思是调用系统函数，32是ASCII码值，表示空格
        eval(['!rename' 32 origin_name 32 new_name]);

    end
end

File: Code/renamebytime.py
import os, glob, datetime

def renamebytime(filepath, file_extension='.JPG'):
    cwd = os.getcwd()
    os.chdir(filepath)
    imgs = glob.iglob(f'{filepath}/*{file_extension}')
    
    for img in imgs:
        origin_name = os.path.basename(img)
        
        date = os.path.getmtime(img) #檔案的建立時間
        datetimeObj = datetime.datetime.fromtimestamp(date)
        new_name = datetimeObj.strftime(f'%m%d_%H%M%S{file_extension}')
        
        a = 1
        while os.path.exists(f'{filepath}/{new_name}'):
            new_name = datetimeObj.strftime(f'%m%d_%H%M%S_{a}{file_extension}')
            a = a + 1
            
        os.rename(origin_name, new_name)
    os.chdir(cwd)